pkg/credentials/service_account_credentials.go

```diff
@@ -63,6 +71,87 @@ func NewCredentialBulder(client client.Client, config *v1.ConfigMap) *Credential
 	}
 }
 
+func (c *CredentialBuilder) CreateStorageSpecSecretEnvs(namespace string, storageKey string,
+	storageSecretName string, overrideParams map[string]string, container *v1.Container) error {
+
+	stype, ok := overrideParams["type"]
+	if ok && stype != "s3" {
+		return errors.New("only S3 type storage is currently supported with storage spec")
```

Comment:
I think this should be done in the validation webhook instead

Reply:
I added the Storage Spec check for these 2 use cases on all the current framework. I also abstracted the framework uri validation function since most framework only validates storageUri only.